Makefile

```diff
@@ -43,6 +43,8 @@ MLRUN_CACHE_DATE ?= $(shell date +%s)
 # 2. add the --cache-from flag to the docker build
 # 3. docker tag and push (also) the (updated) cache image
 MLRUN_DOCKER_CACHE_FROM_TAG ?=
+# if MLRUN_NO_CACHE passed we don't want to use cache, this is mainly used for cleaner if statements
+MLRUN_USE_CACHE := $(if $(MLRUN_NO_CACHE),,true)
```

Comment:
The order I was trying to keep in the Makefile is that the first block of variables are ones that a user might want to override (therefore most of them are `?=`) and the second block is of computed variables (therefore most of them are `:=`)
Doesn't have to be part of this PR but probably this convention should be backed by a comment in the Makefile itself to help people understand where to put new things